superchain/superchain.go

```diff
@@ -436,10 +461,10 @@ type SuperchainConfig struct {
 	SuperchainConfigAddr *Address `yaml:"superchain_config_addr,omitempty"`
 
 	// Hardfork Configuration
-	CanyonTime  *uint64 `yaml:"canyon_time,omitempty"`
-	DeltaTime   *uint64 `yaml:"delta_time,omitempty"`
-	EcotoneTime *uint64 `yaml:"ecotone_time,omitempty"`
-	FjordTime   *uint64 `yaml:"fjord_time,omitempty"`
+	canyonTime  *uint64 `yaml:"canyon_time,omitempty"`
+	deltaTime   *uint64 `yaml:"delta_time,omitempty"`
+	ecotoneTime *uint64 `yaml:"ecotone_time,omitempty"`
+	fjordTime   *uint64 `yaml:"fjord_time,omitempty"`
```

Comment:
Making these private so that they can still be specified and defaulted to, but so that consumers are not tempted to try and read them (they should read from the specific chain which will have the values overridden if appropriate).

Reply:
I believe yaml, json etc marshaling code needs fields to be public (because of reflection).

Reply:
Good point. I think we can get around that by writing a custom `Unmarshal` method. Worth it IMO to avoid confusion.

Reply:
We should also add a comment above that these are potentially overridden by chain-specific values.

Reply:
Good idea. We could even create a struct like `ForkTimes` with all the timestamps, embed this into the chain config and reuse it for marshaling here.